mlir/include/mlir/Dialect/Rock/IR/RockOps.td

```diff
@@ -63,17 +63,17 @@ class IndexArrayLength<int n> : ConfinedAttr<IndexArrayAttr, [ArrayCount<n>]>;
 class Rock_Conv2DOpBase<string mnemonic, list<Type> inputTypes=[F32, F16, BF16], list<Type> outputTypes=[F32, F16, BF16]> :
     Rock_Op<mnemonic, [DeclareOpInterfaceMethods<RockGemmWrapperInterface>,
                        DeclareOpInterfaceMethods<RockConvInterface>]>{
-    dag commonConvArgs = (ins TensorOrMemRefRankOf<inputTypes, [5]>:$filter,
-                   TensorOrMemRefRankOf<inputTypes, [5]>:$input,
-                   TensorOrMemRefRankOf<outputTypes, [5]>:$output,
-                   StrAttr:$arch,
-                   Rock_GemmFeaturesAttr:$features,
-                   OptionalAttr<I32Attr>:$derivedBlockSize,
-                   OptionalAttr<I32Attr>:$gridSize,
-                   IndexArrayLength<4>:$padding,
-                   IndexArrayLength<2>:$strides,
-                   IndexArrayLength<2>:$dilations,
-                   OptionalAttr<RockTuningParamAttrInterface>:$params);
+    dag commonConvArgs = (ins Arg<TensorOrMemRefRankOf<inputTypes, [5]>, "filter", [MemRead]>:$filter,
+                              Arg<TensorOrMemRefRankOf<inputTypes, [5]>, "input", [MemRead]>:$input,
```

Comment:
This isn't correct.

The backwards data operation, for example, writes to the `"input"` tensor, while the backwards weight one writes to `"filter"`

Reply:
Makes sense. So, shall we assign `[MemRead, MemWrite]` for both `filter` and `input` or do you have another idea?

Reply:
See `DeclareOpInterfaceMethods<MemoryEffectsOpInterface>`

Reply:
Hi @krzysz00, Can I ask you to double check the following? 

https://github.com/ROCm/rocMLIR/blob/9e59554827a86467b0f99eb05d5767853820db1e/mlir/lib/Dialect/Rock/IR/RockDialect.cpp#L697-L730

Reply:
Conv2DBwdWeightOp; getFilter (read, write)

Reply:
done